### PR TITLE
Client Telemetry: Fixes consistency level issue

### PR DESCRIPTION
**Pull request by sourabh1007**
## Description

1. Right now, if user is not sending consistency level in the request but sending request option with other configuration. Telemetry is logging Consistency level as "Strong" which is wrong. So fixing it as part of this PR.
2. Optimizing getting consistency level information in telemetry by removing blocking call.
3. Add consistency related tests in client telemetry.
4. Making sure that if user is not passing consistency level (in request or client), it will be stored as NULL in telemetry data.

## Type of change

- [] Bug fix (non-breaking change which fixes an issue)

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Telemetry/OperationInfo.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ internal OperationInfo(string metricsName, string unitName)
             {
                 this.GreaterThan1Kb = responseSizeInBytes > ClientTelemetryOptions.OneKbToBytes;
             }
-            this.Consistency = OperationInfo.GetConsistencyString(consistency);
+            this.Consistency = consistency;
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Telemetry/OperationInfo.cs:65]: It seems we are not uppercasing anymore, should Java SDK do the same change?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/ClientTelemetryTests.cs [hunk 2]
```diff
@@ -352,18 +371,22 @@ private static ItemBatchOperation CreateItem(string itemId)
...
+            if (consistency.HasValue)
             {
-                this.cosmosClient = this.cosmosClientBuilder.WithConnectionModeDirect().Build();
+                this.cosmosClientBuilder = this.cosmosClientBuilder.WithConsistencyLevel(consistency.Value);
             }
 
-            if (mode == ConnectionMode.Gateway)
+            if (mode == ConnectionMode.Direct)
+            {
+                this.cosmosClient = this.cosmosClientBuilder.WithConnectionModeDirect().Build();
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/ClientTelemetryTests.cs:383]: Why is this needed? SDK is Direct by default

File: Microsoft.Azure.Cosmos/src/Handler/TelemetryHandler.cs [hunk 3]
```diff
@@ -38,7 +38,7 @@ public TelemetryHandler(ClientTelemetry telemetry)
                                 databaseId: request.DatabaseId,
                                 operationType: request.OperationType,
                                 resourceType: request.ResourceType,
-                                consistencyLevel: this.GetConsistencyLevel(request),
+                                consistencyLevel: request.Headers[Documents.HttpConstants.HttpHeaders.ConsistencyLevel],
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Handler/TelemetryHandler.cs:41]: just to be safe do we need cover for nullable headers?

**Review by kirankumarkolli — APPROVED** [review]
:shipit:

File: Microsoft.Azure.Cosmos/src/Telemetry/OperationInfo.cs (hunk 1 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Telemetry/OperationInfo.cs:65]: yes...I saw java code. There _overWiredValue_ for ConsistencyLevel is in camelcase so it needs to be fixed in java wherever it is getting changed to uppercase (or getting Enum text as it is, instead of its value).

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/ClientTelemetryTests.cs (hunk 2 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/ClientTelemetryTests.cs:383]: I think intention was here to initialize according to the given connection mode and we are passing it as part of test cases.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/ClientTelemetryTests.cs:383]: The point is, it is not needed, so we shouldn't be adding it. The client is already Direct by default, the if below is to change it to gateway if if's needed

**Review by ealsur — DISMISSED** [review]
Please sync with the Java team in case any casing on the consistency is needed on their side, so both SDKs use the same casing.

**Review by kirankumarkolli — APPROVED** [review]

**Review by j82w — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Telemetry/OperationInfo.cs [hunk 4]
```diff
@@ -31,7 +31,7 @@ internal sealed class OperationInfo
         internal string Resource { get; }
 
         [JsonProperty(PropertyName = "consistency")]
-        internal string Consistency { get; }
+        internal string Consistency { get; set; }
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Telemetry/OperationInfo.cs:34]: we don't need this set, right?

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Telemetry/OperationInfo.cs:34]: yes we don't need it now. removing it.

**Review by ealsur — APPROVED** [review]

**Review by j82w — APPROVED** [review]
